File: BaseBackend/src/main/java/com/common/dto/request/RegisterRequest.java
package com.common.dto.request;

import com.common.utilities.SharedConstants;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

/**
 * DTO yêu cầu đăng ký tài khoản & công ty mới.
 *
 * @created_at 2026-07-31
 * @author txhoan
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class RegisterRequest {

    @NotBlank(message = "{auth.tenant_name_required}")
    String tenantName;

    @NotBlank(message = "{auth.tenant_code_required}")
    String tenantCode;

    @NotBlank(message = "{auth.username_required}")
    @Size(min = 3, max = 50)
    String username;

    @NotBlank(message = "{auth.email_required}")
    @Email(message = "{auth.email_invalid}")
    String email;

    @NotBlank(message = "{auth.password_required}")
    @Pattern(regexp = SharedConstants.PASSWORD_PATTERN, message = "{auth.password_invalid}")
    String password;

    String firstName;

    String lastName;

    @Pattern(regexp = SharedConstants.PHONE_PATTERN, message = "{auth.phone_invalid}")
    String phoneNumber;
}

File: be/.Net/Swe.BL/DTO/Request/RegisterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Swe.BL.DTO.Request;

/// <summary>
/// DTO yêu cầu đăng ký tài khoản & công ty mới.
/// </summary>
/// <created_at>2026-08-03</created_at>
/// <author>txhoan</author>
public class RegisterRequest
{
    [Required(ErrorMessageResourceType = typeof(Swe.Common.Resources.ResourcesVN), ErrorMessageResourceName = "TenantNameRequired")]
    public string TenantName { get; set; } = string.Empty;

    [Required(ErrorMessageResourceType = typeof(Swe.Common.Resources.ResourcesVN), ErrorMessageResourceName = "TenantCodeRequired")]
    public string TenantCode { get; set; } = string.Empty;

    [Required(ErrorMessageResourceType = typeof(Swe.Common.Resources.ResourcesVN), ErrorMessageResourceName = "UsernameRequired")]
    [MinLength(3, ErrorMessageResourceType = typeof(Swe.Common.Resources.ResourcesVN), ErrorMessageResourceName = "UsernameMinLength")]
    [MaxLength(50, ErrorMessageResourceType = typeof(Swe.Common.Resources.ResourcesVN), ErrorMessageResourceName = "UsernameMaxLength")]
    public string Username { get; set; } = string.Empty;

    [Required(ErrorMessageResourceType = typeof(Swe.Common.Resources.ResourcesVN), ErrorMessageResourceName = "EmailRequired")]
    [EmailAddress(ErrorMessageResourceType = typeof(Swe.Common.Resources.ResourcesVN), ErrorMessageResourceName = "EmailInvalid")]
    public string Email { get; set; } = string.Empty;

    [Required(ErrorMessageResourceType = typeof(Swe.Common.Resources.ResourcesVN), ErrorMessageResourceName = "PasswordRequired")]
    public string Password { get; set; } = string.Empty;

    public string? FirstName { get; set; }

    public string? LastName { get; set; }

    public string? PhoneNumber { get; set; }
}
